R/getNeatOrder.R

```diff
@@ -0,0 +1,173 @@
+#' Sorting by radial theta angle
+#' 
+#' @description \code{getNeatOrder} sorts already ordinated data by the radial theta angle.
+#' This method is useful for organizing data points based on their angular 
+#' position in a 2D space, typically after an ordination technique such as PCA or NMDS 
+#' has been applied. 
+#' 
+#' The function takes in a matrix of ordinated data, optionally 
+#' centers the data using specified methods (\code{mean}, \code{median}, or \code{none}), and then calculates 
+#' the angle (theta) for each point relative to the centroid. The data points are then 
+#' sorted based on these theta values in ascending order. 
+#' 
+#' One significant application of this sorting method is in plotting heatmaps. 
+#' By using radial theta sorting, the relationships between data points can be preserved 
+#' according to the ordination method's spatial configuration, rather than relying on 
+#' hierarchical clustering, which may distort these relationships. This approach 
+#' allows for a more faithful representation of the data's intrinsic structure as captured 
+#' by the ordination process.
+#' 
+#' @param x A matrix containing the ordinated data to be sorted. Columns should represent the principal components (PCs) and rows should represent the entities being analyzed (e.g. features or samples). There should be 2 columns only representing 2 PCs.
+#' @param centering A single \code{character} value specifying the method to center the data. Options are \code{mean}, \code{median}, or \code{none} if your data is already centered. (default: method = \code{mean})
+#' @param ... Additional arguments passed to other methods.
+#' @return A \code{character} vector of row names in the sorted order.
+#' 
+#' @details 
+#' It's important to note that the \pkg{sechm} package does actually have the functionality for plotting a heatmap using this radial theta angle ordering, though only by using an MDS ordination. 
+#' 
+#' This functionality can be found at:
+#' 
+#'  \url{https://bioconductor.org/packages/3.18/bioc/vignettes/sechm/inst/doc/sechm.html#row-ordering}. 
+#'  
+#' That being said, the \code{getNeatOrder} function is more modular and separate to the plotting, and 
+#' can be applied to any kind of ordinated data which can be valuable depending on the use case.
+#' 
+#' The below paper outlines the NeatMap method in more detail:
+#' 
+#' "Rajaram & Oono (2010)" - https://doi.org/10.1186/1471-2105-11-45.
```

Comment:
Also this could be added with (Rajaram & Oono (2010) NeatMap - non-clustering heat map alternatives in R)[https://doi.org/10.1186/1471-2105-11-45]

--> Check the manual age from man folder